[main ui] Properly handle XDG format specifications for Linux systems

RodoMa92 opened the pull request:
This fixes the issue where the emulator was writing to the root directory instead of the standard XDG format folders (which is typically $HOME/.local/share/).

Gliniak commented:
https://github.com/xenia-canary/xenia-canary/blob/canary_experimental/src/xenia/base/filesystem_posix.cc#L57
 
 This is where it should be done

RodoMa92 commented:
@Gliniak Should be good now, it was already handled but the default on starting always in portable mode was the problem. I left that setting on only for Windows platform, since that assumption is typically non true for POSIX subsystems.

RodoMa92 committed:
[main ui] Properly handle XDG format specifications for Linux systems.

The issue was stemming from the fact that by default Xenia is running in portable mode, as in use the same root directory as the executable for its datafiles.

This default is an issue when using this software from a POSIX platform, since a lot of the times the location where the binary resides is not writable by default.

Therefore, force portable mode to be disabled on non Windows platforms.

Margen67 commented on src/xenia/app/xenia_main.cc:
```diff
@@ -95,9 +95,15 @@ UPDATE_from_bool(mount_cache, 2024, 8, 31, 20, false);
 DEFINE_transient_path(target, "",
                       "Specifies the target .xex or .iso to execute.",
                       "General");
+#ifndef XE_PLATFORM_WIN32
+DEFINE_transient_bool(portable, false,
+                      "Specifies if Xenia should run in portable mode.",
+                      "General");
+#else
 DEFINE_transient_bool(portable, true,
                       "Specifies if Xenia should run in portable mode.",
                       "General");
+#endif
```
Will this work?
```suggestion
DEFINE_transient_bool(portable,
#ifndef XE_PLATFORM_WIN32
                      false,
#else
                      true,
#endif
                      "Specifies if Xenia should run in portable mode.",
                      "General");
```

RodoMa92 replied:
Good point, I'll test it later.

Gliniak replied:
ping

RodoMa92 replied:
Thanks, I forgot :P

RodoMa92 replied:
Nope, this doesn't work sadly :(
